### PR TITLE
OpenFOAM foss 2015.05 flavour: needed for Centos6 on Haswell

### PR DESCRIPTION
**Pull request by sbridi**
OpenFOAM built with foss 2015.05 needed for Centos6 on Haswell

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by sbridi** [comment]
Added some missing pieces can we recheck?

File: easybuild/easyconfigs/f/flex/flex-2.5.39-foss-2015.05.eb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Built with EasyBuild version 2.1.0 on 2015-05-21_15-38-14
```
Comment (boegel) [inline easybuild/easyconfigs/f/flex/flex-2.5.39-foss-2015.05.eb]: pls remove

File: easybuild/easyconfigs/f/flex/flex-2.5.39-foss-2015.05.eb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+toolchain = {'name': 'foss', 'version': '2015.05'}
+toolchainopts = {'pic': True}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = ['http://prdownloads.sourceforge.net/%(namelower)s']
+
+moduleclass = 'lang'
+
+# Build statistics
```
Comment (boegel) [inline easybuild/easyconfigs/f/flex/flex-2.5.39-foss-2015.05.eb]: pls remove (whole block)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/40cdd0c7707a517a4bec for a full test report.

**Comment by sbridi** [comment]
Test report by @sbridi
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
Linux centos 6.6, Intel(R) Xeon(R) CPU E5-2603 v3 @ 1.60GHz, Python 2.6.6
See https://gist.github.com/8266c7f3a017a1ef9a8a for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (5 easyconfigs in this PR)
Linux red hat enterprise linux computenode 6.5, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.6.6
See https://gist.github.com/77d73c7cab65538948cc for a full test report.

**Comment by boegel** [comment]
Going in, thanks @sbridi!

**Comment by boegel** [comment]
@sbridi: and welcome to the wonderful world of EasyBuild, looking forward to more contributions in the future!

**Comment by boegel** [comment]
@sbridi: you mentioned having issues with this build on IRC w.r.t. `openFoam` not being there, but the OpenFOAM easyblock is making sure it's there, via the sanity check:

```
$ module load OpenFOAM/2.3.1-foss-2015.05
$ file $EBROOTOPENFOAM/OpenFOAM-2.3.1/bin/paraFoam
/gpfs/scratch/projects/project_gpilot/vsc40023/software/OpenFOAM/2.3.1-foss-2015.05/OpenFOAM-2.3.1/bin/paraFoam: POSIX shell script text executable
$ ls -l $EBROOTOPENFOAM/OpenFOAM-2.3.1/bin/paraFoam
-rwxr-xr-x 1 vsc40023 vsc40023 6934 Dec  4  2014 /gpfs/scratch/projects/project_gpilot/vsc40023/software/OpenFOAM/2.3.1-foss-2015.05/OpenFOAM-2.3.1/bin/paraFoam
```

Looks fine, am I missing something?

**Comment by sbridi** [comment]
What is missing is not the paraFoam executable (which is a bash wrapper that start paraView with the openfoam reader libraries) but the "PV4Readers" libraries  and paraview.
When I do:

module load OpenFOAM/2.3.1-foss-2015.05
source $FOAM_BASH
paraFoam

I get this message:

FATAL ERROR: ParaView reader module libraries do not exist

Please build the reader module before continuing:
cd $FOAM_UTILITIES/postProcessing/graphics/PV4Readers
./Allwclean
./Allwmake

**Comment by boegel** [comment]
So, these don't get built by default? Or maybe because Paraview is not listed as a dependency?

**Comment by sbridi** [comment]
In the ThirdParty directory OpenFOAM carries a version of Paraview which is not built along with OpenFOAM. I don't know if it is best to use this version or link the pv4readers to an external build of paraview. I never built 'by hand' openFOAM so I'm not sure how to proceed.

**Comment by boegel** [comment]
I think both are viable options, but since Paraview is useful by itself, installing it as a separate module and then including it as a dependency with OpenFOAM makes more sense.

@besserox has taken a detailed look at OpenFOAM not too long ago, maybe he has a good view on this?

**Comment by besserox** [comment]
Hi,

OpenFOAM has the bad habits of shipping all its dependencies in a ThirdParty package (with OpenMPI, SCOTCH, ParaView, etc.). In the end, it only complexify the build because their buildsystem does (always) not support a way to use another pre-installed version of the same software. For example, EasyBuild has been fighting around this for MPI and SCOTH dependencies.

As far I remember, EasyBuild does not support building OpenFOAM with ParaView so far. So some work is required here. I would recommend the approach that Kenneth suggested, ie relying on a separate build of ParaView and use it as a dependency.

I don't know the part regarding ParaView in the OpenFOAM buildsystem, but if possible
I would recommend to follow the same logic as for SCOTCH:
- in the EasyBlock, check if SCOTCH is available as a dependency, if yes define the SCOTCH_ROOT environment variable 
https://github.com/hpcugent/easybuild-easyblocks/blob/master/easybuild/easyblocks/o/openfoam.py#L143
- patch the OpenFOAM buildsystem to use SCOTCH_ROOT if it is defined
https://github.com/hpcugent/easybuild-easyconfigs/blob/master/easybuild/easyconfigs/o/OpenFOAM/cleanup-OpenFOAM-2.3.1.patch#L42

I hope this helps.

**Pull request merged**